Fix motorola-ocean, add motorola-river and motorola-channel

valpackett opened the pull request:
![Photo of channel and river running lk2nd](https://github.com/user-attachments/assets/87945f8a-23f8-47c6-990e-2c1d576d50b2)

This should fix https://github.com/msm8916-mainline/lk2nd/issues/551 :)

valpackett committed:
gitignore: ignore *.img not dtbo.img

All the generated ones have long names like 'dtbo-motorola-ocean.img'.

valpackett committed:
Fix motorola-ocean variant support by using separate DTBs per board ID

That's how the stock dtbo img is built. The long single list is *not*
searched in, the bootloader only matches the first two numbers in each
dtb so it was only working for one single variant: the first one in the
list (NA EVT1).

Fixes: https://github.com/msm8916-mainline/lk2nd/issues/551

valpackett committed:
Add Motorola Moto G7 (motorola-river)

valpackett committed:
Add Motorola Moto G7 Play (motorola-channel)

valpackett committed:
Makefile: make clean .PHONY

barni2000 commented:
I think it is possible to make a merged image like stock what contains all the dtbo, can you try it?

valpackett committed:
Makefile: support nested directories for dtbos

Also, generally set the target paths to actual paths including the $OUT.

valpackett committed:
Makefile: ignore errors in clean commands

valpackett committed:
Makefile: dynamically generate proper dependencies for .cfg files

Instead of relying on order of the 'all' target, specify all the actual
dependencies by parsing on the fly
